sources/Interop/D3D12MemoryAllocator/include/Allocation.cs

```diff
@@ -0,0 +1,416 @@
+        /// </para>
+        /// </summary>
+        /// <returns>Whether the memory of the allocation was filled with zeros.</returns>
+        [return: NativeTypeName("BOOL")]
+        public readonly int WasZeroInitialized() => m_PackedData.WasZeroInitialized();
+
+        internal enum Type
+        {
+            TYPE_COMMITTED,
+            TYPE_PLACED,
+            TYPE_HEAP,
+            TYPE_COUNT
+        }
+
+        internal readonly ref _m_Committed_e__Struct m_Committed
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => ref MemoryMarshal.GetReference(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(m_Union.m_Committed), 1));
+        }
+
+        internal readonly ref _m_Placed_e__Struct m_Placed
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => ref MemoryMarshal.GetReference(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(m_Union.m_Placed), 1));
+        }
+
+        internal readonly ref _m_Heap_e__Struct m_Heap
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => ref MemoryMarshal.GetReference(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(m_Union.m_Heap), 1));
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct _Anonymous_e__Union
+        {
+            [FieldOffset(0)]
+            public _m_Committed_e__Struct m_Committed;
+
+            [FieldOffset(0)]
+            public _m_Placed_e__Struct m_Placed;
+
+            [FieldOffset(0)]
+            public _m_Heap_e__Struct m_Heap;
+
+            public struct _m_Committed_e__Struct
+            {
+                public D3D12_HEAP_TYPE heapType;
+            }
+
+            public struct _m_Placed_e__Struct
+            {
+                [NativeTypeName("UINT64")]
+                public ulong offset;
+
+                public NormalBlock* block;
+            }
+
+            public struct _m_Heap_e__Struct
+            {
+                public D3D12_HEAP_TYPE heapType;
+                public ID3D12Heap* heap;
+            }
+        }
+
+        internal partial struct PackedData
+        {
+            uint _bitfield0;
+            uint _bitfield1;
+
+            public readonly new Type GetType() => (Type)m_Type;
+
+            public readonly D3D12_RESOURCE_DIMENSION GetResourceDimension() => (D3D12_RESOURCE_DIMENSION)m_ResourceDimension;
+
+            public readonly D3D12_RESOURCE_FLAGS GetResourceFlags() => (D3D12_RESOURCE_FLAGS)m_ResourceFlags;
+
+            public readonly D3D12_TEXTURE_LAYOUT GetTextureLayout() => (D3D12_TEXTURE_LAYOUT)m_TextureLayout;
+
+            [return: NativeTypeName("BOOL")]
+            public readonly int WasZeroInitialized() => (int)m_WasZeroInitialized;
+
+            public void SetType(Type type)
+            {
+                uint u = (uint)type;
+                D3D12MA_ASSERT(u < (1u << 2));
```

Comment:
This, and all other asserts, should be: `D3D12MA_ASSERT((D3D12MA_DEBUG_LEVEL > 0) && (expr))`
```suggestion
                D3D12MA_ASSERT((D3D12MA_DEBUG_LEVEL > 0) && (u < (1u << 2)));
```

Doing this ensures that `(expr)` (`u < (1u << 2)` in this case) can be fully elided from codegen when asserts are disabled.
Otherwise, the JIT could decide to evaluate it anyways for certain side effects

Reply:
Oh, right, cool! 😄
Fixed in 2acd09e47ffaf58cf960801a5f06081161a44b7e for all non-constant assertions.